pkgs/servers/mautrix-telegram/default.nix

```diff
@@ -60,6 +60,8 @@ python.pkgs.buildPythonPackage rec {
     prometheus-client
     # sqlite
     aiosqlite
+    # proxy support
+    pysocks
```

Comment:
But upstream says `python-socks[asyncio]`? https://github.com/mautrix/telegram/blob/806eea53eb1ad85a33c5e0a1261edc7a294dc251/optional-requirements.txt#L28

Reply:
Thanks for pointing that out, I have missed that file.

Honestly I am actually not sure which one should be there. Upstream also says pysocks in the config file here: https://github.com/mautrix/telegram/blob/master/mautrix_telegram/example-config.yaml#L644
... That's where I got it from.

On top of that, documentation doesn't mention any socks dependency for telegram at all, while for facebook it says pysocks: https://docs.mau.fi/bridges/python/optional-dependencies.html#mautrix-telegram

The need for this dependency comes from telethon package, that is used for handling the connection, and so it gets even more confusing when you check their optional deps, they have both in there!...  https://github.com/LonamiWebs/Telethon/blob/v1/optional-requirements.txt

After more investigating, **it appears that the library actually supports both python-socks and pysocks**, as you can see here: https://github.com/LonamiWebs/Telethon/blob/v1/telethon/network/connection/connection.py#L173

From github of both projects, it seems that pysocks is significantly more popular, however it wasn't touched since 2019, while python-socks a little bit more recent, but there is not a huge activity either.

I am leaving the decision up to you, which library you want to use. If you want python-socks, I can update the PR.

Reply:
I have no strong preference, as long as it works for you it's fine.

Reply:
I have verified manually that it works, the bridge contacts telegram servers through http_proxy, I was able to log-in and send and receive messages just fine